operation/manager.go

```diff
@@ -86,13 +87,19 @@ type IsPoweredInterface interface {
 }
 
 // WaitTillNotPowered waits until IsPowered returns false.
-func (sm *SingleOperationManager) WaitTillNotPowered(ctx context.Context, pollTime time.Duration, powered IsPoweredInterface) error {
+func (sm *SingleOperationManager) WaitTillNotPowered(ctx context.Context, pollTime time.Duration, powered IsPoweredInterface,
+	stop func(context.Context, map[string]interface{}) error,
+) error {
 	return sm.WaitForSuccess(
 		ctx,
 		pollTime,
 		func(ctx context.Context) (bool, error) {
+			var errStop error
 			res, err := powered.IsPowered(ctx, nil)
-			return !res, err
+			if err != nil {
+				errStop = stop(ctx, map[string]interface{}{})
```

Comment:
I actually think stop should be passed into WaitForSuccess as a cleanupFunc and called there (maybe as a deferred function?). There is a chance that if the context is cancelled during the `SelectContextOrWait` part in WaitForSuccess, the stop function will never get called.

Reply:
Agreed, added the stop function as a defer to ensure correct clean up